### PR TITLE
RATIS-2165. Do not include RaftClientRequest Message in ReadIndexRequest

### PR DESCRIPTION
**Pull request by 133tosakarin**
## What changes were proposed in this pull request?

For the readIndexRequestProto, we don't need to set message member 
In readIndex, we did not use clientRequest.message, so we do not need to serialize the message

## What is the link to the Apache JIRA

https://issues.apache.org/jira/browse/RATIS-2165

### CODE REVIEW

File: ratis-client/src/main/java/org/apache/ratis/client/impl/ClientProtoUtils.java [hunk 1]
```diff
@@ -204,9 +204,13 @@ static ByteBuffer toRaftClientRequestProtoByteBuffer(RaftClientRequest request)
   }
 
   static RaftClientRequestProto toRaftClientRequestProto(RaftClientRequest request) {
+    return toRaftClientRequestProto(request, true);
+  }
+
+  static RaftClientRequestProto toRaftClientRequestProto(RaftClientRequest request, boolean with_msg) {
```
Comment (OneSizeFitsQuorum) [inline ratis-client/src/main/java/org/apache/ratis/client/impl/ClientProtoUtils.java:210]: do not use `with_msg`, at least it should be `withMsg`

**Comment by 133tosakarin** [comment]
@szetszwo 

These failed tests seem to meet expectations

**Review by szetszwo — APPROVED** [review]
+1 the change looks good.

**Comment by szetszwo** [comment]
> These failed tests seem to meet expectations

@133tosakarin , you are right that the failures does not seem related; see https://github.com/apache/ratis/pull/1156 .  Let's merge this without further retries.

**Pull request merged**